src/main/java/org/jabref/logic/util/strings/EmacsStringManipulator.java

```diff
@@ -0,0 +1,178 @@
+package org.jabref.logic.util.strings;
+import org.jabref.model.util.ResultingEmacsState;
+
+public class EmacsStringManipulator {
+    private enum LetterCase {
+        UPPER,
+        LOWER,
+        CAPITALIZED
+    }
+
+    private enum Direction {
+        NEXT, PREVIOUS
+    }
+
+    private static ResultingEmacsState setNextWordsCase(String text, int pos, LetterCase targetCase) {
+        StringBuilder res = new StringBuilder();
+
+        boolean firstLetter = true;
+        int i = pos;
+        boolean firstLoop = true;
+        StringBuilder newWordBuilder = new StringBuilder();
+        for (; i < text.length(); i++) {
+            // Swallow whitespace
+            while (firstLoop && i < text.length() && !String.valueOf(text.charAt(i)).matches("\\w")) {
+                newWordBuilder.append(text.charAt(i));
+                i++;
+            }
+            if (i >= text.length()) {
+                break;
+            }
+            if (firstLoop) {
+                firstLoop = false;
+            }
+            char currentChar = text.charAt(i);
+            if (String.valueOf(currentChar).matches("\\w")) {
+                switch (targetCase) {
+                    case UPPER:
+                        newWordBuilder.append(Character.toUpperCase(currentChar));
```

Comment:
I searched the code for `LOCALE` for `toUpperCase`, because of https://haacked.com/archive/2012/07/05/turkish-i-problem-and-why-you-should-care.aspx/. I know that JabRef isn't doing it quite right.

Could you please rewrite your code to

1. Extract the string where the case needs to be changed
2. Use `org.jabref.logic.bst.BibtexCaseChanger#changeCase`.

Reason: code-reuse and we are in a BibTeX setting which needs special treatment.

Reply:
JavaDoc in BibtexCaseChanger says:
```
// the following would have to be done if the functionality of CaseChangers would be included here
// However, we decided against it and will probably do the other way round: https://github.com/JabRef/jabref/pull/215#issuecomment-146981624
```

So I will use CaseChangeFormatters (also because the BibtexCaseChanger does not support simple capitalization :-o 😉